util/tasks/threaded_task_runner.cpp

```diff
@@ -90,11 +90,11 @@ void ThreadedTaskRunner::set_thread_count(uint32_t count) {
 		count = MAX_THREADS;
 	}
 	destroy_all_threads();
-	for (uint32_t i = _thread_count; i < count; ++i) {
+	_thread_count = count;
+	for (uint32_t i = 0; i < _thread_count; ++i) {
```

Comment:
I have a feeling that method was partially thought to be implemented incrementally, without destroying all threads. But I must have reverted to destroy them anyways because I didnt bother changing at runtime. Yet part of the incremental way remained.
Adding more threads means creating the extra ones.
Removing threads means "flagging" the extra ones so that they stop by themselves.
Though it seems a bit easier to destroy them all, at the cost of lag.